public/app/core/components/Login/LoginCtrl.tsx

```diff
@@ -104,9 +104,17 @@ export class LoginCtrl extends PureComponent<Props, State> {
     const params = this.props.routeParams;
     // Use window.location.href to force page reload
     if (params.redirect && params.redirect[0] === '/') {
-      window.location.href = config.appSubUrl + params.redirect;
+      if (config.appSubUrl !== '' && !params.redirect.startsWith(config.appSubUrl)) {
```

Comment:
Didn't understand why we still have this logic of looking at redirect querystring parameter? Seems like we use it for the sidemenu: https://github.com/grafana/grafana/blob/master/public/app/core/components/sidemenu/SignIn.tsx#L6